src/kvirc/ui/KviInputEditor.cpp

```diff
@@ -2282,54 +2293,38 @@ void KviInputEditor::completion(bool bShift)
 				}
 			}
 
-			if(bInCommand && !bIsCommand)
-			{
+			if(bInCommand && !bIsCommand && !bIsFunction)
 				completionEscapeUnsafeToken(szMatch); // escape crazy things like Nick\nquit
-			}
 		}
 		else
 		{
 			QString szAll;
 			szMatch = tmp.front();
-			int iWLen = szWord.length();
-			if(szMatch.left(1) == '$')
-				szMatch.remove(0, 1);
-			for(auto szTmpIter : tmp)
+			auto predicate = bIsDir
```

Comment:
That works too.  I try to stay away from lambda captures unless strictly necessary, but since older MSVC requires a `std::function` wrapper, I'd agree with you on it looking better.